src/engine/renderer/tr_shader.cpp

```diff
@@ -5992,209 +5990,154 @@ A second parameter will cause it to print in sorted order
 */
 void R_ShaderList_f()
 {
-	int      i;
-	int      count;
-	shader_t *shader;
-	const char *s = nullptr;
-
-	Log::Notice("-----------------------" );
-
-	if ( ri.Cmd_Argc() > 1 )
-	{
-		s = ri.Cmd_Argv( 1 );
-	}
-
-	count = 0;
-
-	for ( i = 0; i < tr.numShaders; i++ )
-	{
-		if ( ri.Cmd_Argc() > 2 )
-		{
-			shader = tr.sortedShaders[ i ];
-		}
-		else
-		{
-			shader = tr.shaders[ i ];
-		}
-
-		if ( s && !Com_Filter( s, shader->name, false ) )
+	const char *prefix = ri.Cmd_Argc() > 1 ? ri.Cmd_Argv( 1 ) : nullptr;
+
+	using shaderTypeStringMap = std::unordered_map<shaderType_t, std::string>;
+	shaderTypeStringMap shaderTypeStrings;
+	shaderTypeStrings[shaderType_t::SHADER_2D] = "2D";
+	shaderTypeStrings[shaderType_t::SHADER_3D_DYNAMIC] = "3D_DYNAMIC";
+	shaderTypeStrings[shaderType_t::SHADER_3D_STATIC] = "3D_STATIC";
+	shaderTypeStrings[shaderType_t::SHADER_LIGHT] = "LIGHT";
+
+	std::string shaderTypeHeader = "shaderType";
+	size_t maxShaderTypeStringLen = shaderTypeHeader.length();
+
+	for ( const auto& kv : shaderTypeStrings )
+	{
+		maxShaderTypeStringLen = std::max( maxShaderTypeStringLen, kv.second.length() );
+	}
+
+	using shaderSortStringMap = std::unordered_map<shaderSort_t, std::string>;
```

Comment:
The two `using` statements could be removed since the defined type is only used once.

Reply:
I removed the three now useless `using`.